src/appengine/libs/query/datastore_query.py

```diff
@@ -281,14 +281,32 @@ def fetch(self, offset, limit, projection, more_limit):
 
   def fetch_page(self, page, page_size, projection, more_limit):
     """Return the items, total_pages, total_items, and has_more."""
+
+    # Validation check to confirm positive page number.
+    if page > 0:
```

Comment:
Actually, let's just set it to 1 by default to keep the implementation simple. There's no real benefit of showing nothing vs showing just the first page in this case.

Reply:
if we would like to do that, it would be better to that on the front-end side i.e. change the negative requested page number before sending posting it. As if we just handle it here, the page number in the front-end side of the stack will remain the same. So, if user requests for page -5 and we show data for page 1, and then he clicks next. His page number will shift to -4.

Reply:
That's fine. This is not a case that anyone would run into normally. I think we are spending too much engineering effort to solve this here :)

Reply:
yes, I agree. I have updated this to keep 1 by default and added tests.